pkg/relayer/session/claim.go

```diff
@@ -235,3 +240,75 @@ func (rs *relayerSessionsManager) goCreateClaimRoots(
 	failSubmitProofsSessionsCh <- failedClaims
 	claimsFlushedCh <- flushedClaims
 }
+
+// payableProofsSessionTrees returns the session trees that the supplier operator
+// can afford to claim.
+func (rs *relayerSessionsManager) payableProofsSessionTrees(
+	ctx context.Context,
+	sessionTrees []relayer.SessionTree,
+) ([]relayer.SessionTree, error) {
+	logger := rs.logger.With(
+		"supplier_operator_address", sessionTrees[0].GetSupplierOperatorAddress().String(),
+	)
+
+	proofParams, err := rs.proofQueryClient.GetParams(ctx)
+	if err != nil {
+		return nil, err
+	}
+	proofSubmissionFeeCoin := proofParams.GetProofSubmissionFee()
+
+	supplierOperatorBalanceCoin, err := rs.bankQueryClient.GetBalance(
+		ctx,
+		sessionTrees[0].GetSupplierOperatorAddress().String(),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// Sort the session trees by the sum of the claim root to ensure that the
+	// biggest claim roots are claimed first.
+	slices.SortFunc(sessionTrees, func(a, b relayer.SessionTree) int {
+		rootA := a.GetClaimRoot()
+		sumA, errA := smt.MerkleSumRoot(rootA).Sum()
+		if errA != nil {
+			logger.With(
+				"session_id", a.GetSessionHeader().GetSessionId(),
+				"claim_root", fmt.Sprintf("%x", rootA),
+			).Error().Err(errA).Msg("failed to calculate sum of claim root, assuming 0")
+			sumA = 0
+		}
+
+		rootB := b.GetClaimRoot()
+		sumB, errB := smt.MerkleSumRoot(rootB).Sum()
+		if errB != nil {
+			logger.With(
+				"session_id", a.GetSessionHeader().GetSessionId(),
+				"claim_root", fmt.Sprintf("%x", rootA),
+			).Error().Err(errB).Msg("failed to calculate sum of claim root, assuming 0")
+			sumB = 0
+		}
+
+		// Sort in descending order.
+		return int(sumB - sumA)
```

Comment:
Does this function have a unit test to make sure it works as expected?

Reply:
Yes [pkg/relayer/session/session_test.go#L220](https://github.com/pokt-network/poktroll/pull/817/files#diff-4118be5dadfdb12668216464d569818a98ab9694e4e9c9f505469317d9536516R220)